client/tilespec.cpp

```diff
@@ -5224,13 +5224,14 @@ static int fill_grid_sprite_array(
           worked[i] = (NULL != tile_worked(tile));
         }
       }
-        if (tile) {
-          { // Draw city grid for main citymap
-            xcity = find_city_near_tile(tile);
-            if (xcity && xcity->client.city_opened) {
-              ADD_SPRITE_SIMPLE(t->sprites.grid.selected[pedge->type]);
-            }
+      if (tile) {
+        { // Draw city grid for main citymap
+          xcity = find_city_near_tile(tile);
```

Comment:
I think it might still fail because `find_city_near_tile` always returns the closest city?

Reply:
comment says that:
   Find the "best" city/settlers to associate with the selected tile.
     a.  If a visible city is working the tile, return that city.
     b.  If another player's city is working the tile, return NULL.
     c.  If any selected cities are within range, return the closest one.
     d.  If any cities are within range, return the closest one.
     e.  If any active (with color) settler could work it if they founded a
         city, choose the closest one (only if punit != NULL).
     f.  If any settler could work it if they founded a city, choose the
         closest one (only if punit != NULL).
     g.  If nobody can work it, return NULL.

Reply:
I tried with some cities near and was working as expected but maybe I was lucky

Reply:
ok let's merge and reopen the issue if it's still there

Reply:
or maybe I jsut iterate to be sure